From a7e7194da96736d47c2291e5ad3176c161f16a4b Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Sun, 12 Aug 2018 12:02:43 +0800
Subject: [PATCH 4383/5242] MLK-19158-3 gpu: imx: lcdif: fix output order for
 16bpp BGR formats

commit  7b35144583235352fd219eea2c9f06342e1d7d2d from
https://source.codeaurora.org/external/imx/linux-imx.git

The 16bpp BGR order pixel formats 'DRM_FORMAT_ABGR1555' and
'DRM_FORMAT_XBGR1555' also require to be re-ordered to RGB
order for display, just like the format 'DRM_FORMAT_BGR565'
does.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit f5cc4f4699570fe697d21cb47c54aa91b82c8458)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/lcdif/lcdif-common.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/lcdif/lcdif-common.c b/drivers/gpu/imx/lcdif/lcdif-common.c
index f7b887c..86b1069 100644
--- a/drivers/gpu/imx/lcdif/lcdif-common.c
+++ b/drivers/gpu/imx/lcdif/lcdif-common.c
@@ -317,7 +317,9 @@ int lcdif_set_pix_fmt(struct lcdif_soc *lcdif, u32 format)
 		ctrl1 |= CTRL1_SET_BYTE_PACKAGING(0xf);
 
 		/* 'BGR' order */
-		if (format == DRM_FORMAT_BGR565)
+		if (format == DRM_FORMAT_BGR565		||
+		    format == DRM_FORMAT_ABGR1555	||
+		    format == DRM_FORMAT_XBGR1555)
 			writel(CTRL2_ODD_LINE_PATTERN(0x5) |
 			       CTRL2_EVEN_LINE_PATTERN(0x5),
 			       lcdif->base + LCDIF_CTRL2 + REG_SET);
-- 
1.7.9.5

